### ISSUE TITLE
Create documentation for ALL of the Sphinx-RST format

### ISSUE THREAD

**Issue by radix** [writethedocs/www#40]
Right now, learning to write Sphinx-style RST documentation requires referring to several hard-to-find documents spread out all across the Internet. It would be lovely if there were a single document describing everything possible with ReStructuredText and Sphinx, including basic syntax and full documentation for all roles available in a Sphinx environment, ideally in a single web page that's easily searchable.

**Comment by gwideman** [comment]
+100000.
For others who might stumble in here:

-- WriteTheDocs implies using the Sphinx program to convert markup-files to HTML (and other formats).
-- Sphinx (http://sphinx-doc.org/intro.html), apparently created by the WTD folks, understands reStructuredText markup language (http://docutils.sourceforge.net/rst.html), whose parser is part of a python package called Docutils (http://docutils.sourceforge.net/index.html)

If you just want to understand the capabilities and syntax involved in reStructuredText and Sphinx, then the following references seem relevant:
-- List of docs: http://docutils.sourceforge.net/rst.html#reference-documentation
-- Quick reference: http://docutils.sourceforge.net/docs/user/rst/quickref.html
-- Cheat sheet: http://docutils.sourceforge.net/docs/user/rst/cheatsheet.html (note: confusingly, several important syntax elements are shown within the Grid table and Simple table examples, so the concepts and syntax of tables and various block and list features are all muddled together.)
-- Sphinx TOC: http://sphinx-doc.org/contents.html  See sections "Sphinx Markup Constructs" and "Sphinx Domains" for ways in which Sphinx extends reStructuredText.

[Edit: corrected the spelling of 'sphinx']

**Comment by webmaven** [comment]
:+1:

**Comment by xmo-odoo** [comment]
> Sphinx (http://sphinx-doc.org/intro.html), apparently created by the WTD folks

FWIW Sphinx (with an _I_) was created and is maintained by Georg Brandl under the Pocoo umbrella. WTD is Eric Holscher and Troy Howard's project, as a continuation to Eric Holscher's "ReadTheDocs" project. It uses Sphinx, but is not the source of Sphinx.

IIRC Sphinx was originally created to allow writing better documentation for [Python](https://docs.python.org/3/).

Part of the reStructuredText documentation is also available on Sphinx's official website with somewhat more readable formatting as the [primer/quickref](http://sphinx-doc.org/rest.html)

**Comment by ericholscher** [comment]
This is something that I am strongly considering FWIW -- The RST docs are pretty bad at the moment, and the community there seems really resistant to change. I am quite busy for the next couple weeks, but I would happily endorse and help out with an effort to document RST in Sphinx with a better interface.

**Comment by gwideman** [comment]
@xmo-odoo Thanks for the clarifications. I think I was confused by the interlinking between WTD docs and RST docs, and thereby misunderstood what is subsidiary to what (or not).

(And do the '[)' brackets around 'Python' mean anything significant?)

Your comment has led me to some additional documents:

-- **reStructuredText Markup Specification**: http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html
-- **The Docutils Document Tree**: http://docutils.sourceforge.net/docs/ref/doctree.html

The Markup Specification seems particularly useful as a "ground truth" reference.

**Comment by xmo-odoo** [comment]
@gwideman 

>  (And do the '[)' brackets around 'Python' mean anything significant?)

Only that I had mis-formatted my Markdown link and hadn't proofread the rendering, I've fixed it.

**Comment by asmeurer** [comment]
There's also http://openalea.gforge.inria.fr/doc/openalea/doc/_build/html/source/sphinx/rest_syntax.html, which has some stuff not in the docutils docs (like the sidebar directive).

**Comment by gwideman** [comment]
@asmeurer  Plant architecture modeling -- awesome! Oh, and highly distributed documentation of RST -- not so awesome! But does demonstrate the need to get the authoritative version all in one place.

**Comment by barbara-sfx** [comment]
I am totally new to Sphinx and RST, and have to get myself up to speed quickly as I have proposed we convert our current set of docs to use these tools and move to readthedocs platform. (This is based on feature requests from management and engineers.) 

I'm usually good at teaching myself new stuff quickly, but I think I'm going to have trouble here. One example: I've looked at some current repos for orgs that use sphinx, and several have a map.rst.inc file. I can look at it and can probably figure out what it is, but I can't find any reference to it in any of the docs I have looked at so far. So I'm afraid that I really won't be able to develop a doc set/structure and build process without some in-person instructions/hand-holding. There's probably a lot of other stuff to know and I am sure I currently don't know what I don't know.

I admit I have only skimmed the docs so far, and will be more knowledgeable in general after I have stepped through all the tutorials etc, but I am happy to have discovered this issue and will review the docs named here as well while I educate myself. (Docutils is now added to my to-learn-about list.) Any additional pointers to more docs would be much appreciated. Thanks.
